pkg/api/http/http.go

```diff
@@ -1827,30 +1832,35 @@ func (a *api) onBulkPublish(reqCtx *fasthttp.RequestCtx) {
 
 // validateAndGetPubsubAndTopic takes input as request context and returns the pubsub interface, pubsub name, topic name,
 // or error status code and an ErrorResponse object.
-func (a *api) validateAndGetPubsubAndTopic(reqCtx *fasthttp.RequestCtx) (pubsub.PubSub, string, string, int, *ErrorResponse) {
+func (a *api) validateAndGetPubsubAndTopic(reqCtx *fasthttp.RequestCtx) (pubsub.PubSub, string, string, error) {
+	var err error
+	pubsubName := reqCtx.UserValue(pubsubnameparam).(string)
+	pubsubType := string(contribMetadata.PubSubType)
+	metadata := getMetadataFromFastHTTPRequest(reqCtx)
+
 	if a.pubsubAdapter == nil {
-		msg := NewErrorResponse("ERR_PUBSUB_NOT_CONFIGURED", messages.ErrPubsubNotConfigured)
-		return nil, "", "", nethttp.StatusBadRequest, &msg
+		err = apierrors.NotConfigured(pubsubName, pubsubType, metadata, codes.FailedPrecondition, nethttp.StatusBadRequest, "ERR_PUBSUB_NOT_CONFIGURED", kiterrors.CodePrefixPubSub+kiterrors.CodeNotConfigured)
+		return nil, "", "", err
 	}
 
-	pubsubName := reqCtx.UserValue(pubsubnameparam).(string)
 	if pubsubName == "" {
-		msg := NewErrorResponse("ERR_PUBSUB_EMPTY", messages.ErrPubsubEmpty)
-		return nil, "", "", nethttp.StatusNotFound, &msg
+		err = apierrors.PubSubNameEmpty(pubsubName, pubsubType, metadata)
+		return nil, "", "", err
 	}
 
 	thepubsub, ok := a.universal.CompStore().GetPubSub(pubsubName)
 	if !ok {
-		msg := NewErrorResponse("ERR_PUBSUB_NOT_FOUND", fmt.Sprintf(messages.ErrPubsubNotFound, pubsubName))
-		return nil, "", "", nethttp.StatusNotFound, &msg
+		err = apierrors.NotFound(pubsubName, pubsubType, metadata, codes.InvalidArgument, nethttp.StatusNotFound, "ERR_PUBSUB_NOT_FOUND", kiterrors.CodePrefixPubSub+kiterrors.CodeNotFound)
+		return nil, "", "", err
 	}
 
 	topic := reqCtx.UserValue(wildcardParam).(string)
 	if topic == "" {
-		msg := NewErrorResponse("ERR_TOPIC_EMPTY", fmt.Sprintf(messages.ErrTopicEmpty, pubsubName))
-		return nil, "", "", nethttp.StatusNotFound, &msg
+		err = apierrors.PubSubTopicEmpty(pubsubName, pubsubType, metadata)
+		return nil, "", "", err
 	}
-	return thepubsub.Component, pubsubName, topic, nethttp.StatusOK, nil
+
+	return thepubsub.Component, pubsubName, topic, err
```

Comment:
```suggestion
	return thepubsub.Component, pubsubName, topic, nil
```

Reply:
if err is nil, it will return nil. I can still change it if desired